coverage/src/BamView-WebComponent.js

```diff
@@ -303,26 +190,26 @@ class BamViewChart extends HTMLElement {
         if (this.resizeObserver) {
             this.resizeObserver.disconnect();
         }
+        document.removeEventListener('go-click', this.handleGoClick);
+        document.removeEventListener('search-click', this.handleSearchClick);
     }
     
-    handleGoClick() {
-        let parsedStart, parsedEnd;
-        const chromosome = this.chromosomeInput.value.trim();
-        const startInput = this.startInput.value.trim();
-        const endInput = this.endInput.value.trim();
+    handleGoClick(detail) {
+        const { chromosome, start, end } = detail;
 
         // Validate chromosome number first
         if (!this.isValidChromosome(chromosome)) {
             alert('Invalid chromosome number');
             return;
         }
-        
+
+        let parsedStart, parsedEnd;
         // Check if start and end inputs are non-empty before parsing
-        if (startInput !== "") {
-            parsedStart = parseInt(startInput);
+        if (start !== "") {
```

Comment:
You should be able to get rid of these safety checks once you emit the format described above.